pkg/kubeapiserver/default_storage_factory_builder.go

```diff
@@ -84,26 +86,29 @@ type StorageFactoryConfig struct {
 	EncryptionProviderConfigFilepath string
 }
 
-func (c *StorageFactoryConfig) Complete(etcdOptions *serveroptions.EtcdOptions) (*completedStorageFactoryConfig, error) {
+// Complete completes the StorageFactoryConfig with provided etcdOptions returning CompletedStorageFactoryConfig.
+func (c *StorageFactoryConfig) Complete(etcdOptions *serveroptions.EtcdOptions) (*CompletedStorageFactoryConfig, error) {
 	c.StorageConfig = etcdOptions.StorageConfig
 	c.DefaultStorageMediaType = etcdOptions.DefaultStorageMediaType
 	c.EtcdServersOverrides = etcdOptions.EtcdServersOverrides
 	c.EncryptionProviderConfigFilepath = etcdOptions.EncryptionProviderConfigFilepath
-	return &completedStorageFactoryConfig{c}, nil
+	return &CompletedStorageFactoryConfig{c}, nil
 }
 
-type completedStorageFactoryConfig struct {
+// CompletedStorageFactoryConfig is a wrapper around StorageFactoryConfig completed with etcd options.
+type CompletedStorageFactoryConfig struct {
```

Comment:
do not export this type, it is intentionally private so that it can only be constructed via a call to Complete

Reply:
Ok. Do we want to leave it as is then, making this package never conform to golint rules? I know that golint is not a "gold standard", but maybe there are other options for fixing this issue. Like an introduction of an interface (which on the other hand may introduce extra complexity).

If we just leave it unexported and explicitly returned from the call to Complete, do you want me to add a comment describing that this is intentional?

Reply:
> Do we want to leave it as is then, making this package never conform to golint rules? I know that golint is not a "gold standard", but maybe there are other options for fixing this issue.

I'm vaguely opposed to the current use of golint as a presubmit (it even goes against golint's recommended use - "In short, this tool is not, and will never be, trustworthy enough for its suggestions to be enforced automatically, for example as part of a build process.")

I don't think the complexity of an interface is warranted by the narrow benefit of complying with golint.

> If we just leave it unexported and explicitly returned from the call to Complete, do you want me to add a comment describing that this is intentional?

Let's do that for now.

Reply:
Reverted the change and added the comment.